tests/devices/test_default_clifford.py

```diff
@@ -424,6 +424,54 @@ def test_pl_to_stim(pl_op, stim_op):
     assert wires == " ".join(map(str, stim_op[1]))
 
 
+@pytest.mark.parametrize(
+    ["measurement", "tag"],
+    [
+        (qml.expval(op=qml.Z(1)), None),
+        (qml.expval(op=qml.Y(0) @ qml.X(1)), "expval"),
+        (qml.var(op=qml.X(0)), None),
+        (qml.var(op=qml.X(0) @ qml.Z(1)), "var"),
+        (qml.density_matrix(wires=[1]), None),
+        (qml.density_matrix(wires=[0, 1]), "dm"),
+        (qml.probs(op=qml.Y(0)), None),
+        (qml.probs(op=qml.X(0) @ qml.Y(1)), "probs"),
+        (qml.vn_entropy(wires=[0]), None),
+        (qml.vn_entropy(wires=[1]), "vn_entropy"),
+        (qml.mutual_info(wires0=[1], wires1=[0]), None),
+        (qml.mutual_info(wires0=[0], wires1=[1]), "mi"),
+        (qml.purity(wires=[0]), None),
+        (qml.purity(wires=[1]), "purity"),
+    ],
+)
+def test_snapshot_supported(measurement, tag):
+    """Tests that applying snapshot of measurements is done correctly"""
+
+    def circuit():
+        """Snapshot circuit"""
+        qml.Hadamard(wires=0)
+        qml.Hadamard(wires=1)
+        qml.Snapshot(measurement=qml.expval(qml.Z(0) @ qml.Z(1)))
+        qml.CNOT(wires=[0, 1])
+        qml.Snapshot(measurement=measurement, tag=tag)
```

Comment:
Do we test that the tag has properly been assigned and used?

Reply:
We are testing it [here](https://github.com/PennyLaneAI/pennylane/blob/16205afa10770ab36701c756766eea0f2b3eeeba/tests/devices/test_default_clifford.py#L471) since they tags will be the `keys` of the snapshot dictionary.